kevm-pyk/src/kevm_pyk/foundry.py

```diff
@@ -768,6 +751,32 @@ def foundry_step_node(
             apr_proof.write_proof_data()
 
 
+def foundry_get_apr_proof(
+    foundry_root: Path,
+    test: str,
+    bug_report: bool = False,
+) -> APRProof:
+    proof = foundry_get_proof(foundry_root=foundry_root, test=test, bug_report=bug_report)
+    if not isinstance(proof, APRProof):
+        raise ValueError('Specified proof is not an APRProof.')
+    return proof
+
+
+def foundry_get_proof(
+    foundry_root: Path,
+    test: str,
+    bug_report: bool = False,
+) -> Proof:
+    br = BugReport(Path(f'{test}.bug_report')) if bug_report else None
+    foundry = Foundry(foundry_root, bug_report=br)
+
+    proofs_dir = foundry.out / 'apr_proofs'
+    contract_name, test_name = test.split('.')
+    proof_digest = foundry.proof_digest(contract_name, test_name)
+    proof = Proof.read_proof_data(proofs_dir, proof_digest)
+    return proof
```

Comment:
All the call sites of this have a `Foundry` in hand, so maybe instead of passing in `foundry_root`, you just pass the `Foundry` object in directly?

Construction of a `Foundry` object is expensive, requires reading in all teh Solidity sources and processing them.

Reply:
Fixed.